tests/test_tools.py

```diff
@@ -665,3 +666,121 @@ def test_fragment_retrieval(input, expected):
 def test_month_and_year(input,expected):
     assert(month_and_year(input[0],input[1]) == expected) 
 
+@pytest.mark.parametrize(
```

Comment:
nice!   I would test using dates mostly (i.e., 2020-06-17) and just one test that uses day/month/year to test that pattern for the sake of backwards compatibility, as moving forward we are moving to dates which are easier to keep track of in the database.

Reply:
done